### DISCUSSION TITLE
[Windows] Python command line window when launching streams

### DISCUSSION THREAD

**Discussion by thishead** [streamlink/streamlink-twitch-gui#768] [Q&A] [answered]
it works and my mpv.exe app opens with the stream playing
but i also get an empty black window with this directory on top

C:\Program Files (x86)\Streamlink\Python\python.exe

i upgraded both streamlink and twitch_gui and im trying to figure out how to get rid of that 2nd python window. if i close it it kills my player.

in my settings -streaming - streamlink drop down -  i have noting in custom parameters. i think maybe i deleted that text and shouldnt have. 

id like to keep the empty black window from opening. 

i played around with the other drop down - streamlink python but i got errors when starting the streams from there, i dont think i had the right script info dunno

**Comment by bastimeyer** [comment, marked as answer]
**See the streamlink configuration page on the wiki, which already tells you how to solve your issue:**
https://github.com/streamlink/streamlink-twitch-gui/wiki/Streamlink-configuration

But let me try to answer your question in detail as well.

----

> C:\Program Files (x86)\Streamlink\Python\python.exe

You must have set this custom path in the streaming settings and chosen the `Streamlink (Python)` streaming provider, because Streamlink Twitch GUI [doesn't use the `python.exe` executable](https://github.com/streamlink/streamlink-twitch-gui/blob/v1.11.0/src/config/streaming.json#L50). You don't have to set any custom paths, because the Twitch GUI will automatically resolve the right path for you by looking at the directories in your system's `PATH` environment variable and by using a set of built-in default paths where Streamlink is usually installed.

There are two different streaming provider methods you can choose, `Streamlink` and `Streamlink (Python)`:
The first one, which is the default selection on all OSes [since `v1.10.0`](https://github.com/streamlink/streamlink-twitch-gui/blob/master/CHANGELOG.md#v1100-2020-05-31), only takes the simple [`streamlinkw.exe`](https://github.com/streamlink/streamlink-twitch-gui/blob/v1.11.0/src/config/streaming.json#L24) executable on Windows and the second one lets you choose the [`pythonw.exe`](https://github.com/streamlink/streamlink-twitch-gui/blob/v1.11.0/src/config/streaming.json#L50) python executable and the Streamlink python entry script. If you don't know what you're doing, **just keep the first/default streaming provider and keep the custom path empty**, so that the Twitch GUI can look it up itself.

The streaming providers have been changed a couple of times in recent releases due to changes in the Windows installer for Streamlink and if you've been using the Twitch GUI prior to release `v1.7.0`, [the changelog explicitly mentioned](https://github.com/streamlink/streamlink-twitch-gui/blob/master/CHANGELOG.md#v170-2019-04-05) that all uses of Windows should manually change their streaming provider from an old selection to a new one, which back then had a different name but is now the default selection with the simple `Streamlink` name after another change in [`v1.10.0`](https://github.com/streamlink/streamlink-twitch-gui/blob/master/CHANGELOG.md#v1100-2020-05-31). However, even the old provider, which is now called `Streamlink (Python)`, won't cause a python window to open unless you have accidentally changed something, because this was the entire reason behind this provider selection in the past. But to not make it any more confusing than it already is, just ignore the streaming providers and simply select the default one, as already said.

Now to your issue:

> how to get rid of that 2nd python window

As explained on the wiki page linked above, on Windows, there's a distinction between `python.exe` and `pythonw.exe` (**notice the `w` suffix**). This is also the reason why `streamlink.exe` and `streamlinkw.exe` are built in the Streamlink Windows installer. On the executables without the `w` suffix, a terminal window (CMD window) will always be opened when a python process is started from a non-terminal-window context, like from the Twitch GUI for example. The executables with the `w` suffix prevent this, so they need to be used by applications like Streamlink Twitch GUI.

So in order to fix your issue, select the executable with the `w` suffix or don't select anything at all, because the Twitch GUI will resolve it automatically on its own anyway unless you've made changes to your system.

**Reply by thishead** [reply]
thanks :) removing the the exe path from the streamlink executable box worked. i  dont know why i didnt try that yesterday lol

thanks again
